main/file.c

```diff
@@ -382,83 +331,43 @@ static void sub_go_to_sleep(uint64_t interval_ms){
             esp_deep_sleep_start();
         }
     }
-
-}
-
-static bool check_transmit_period(uint64_t interval_ms){
-
-    bool flag = false;
-
-    for(int i = 0; i < ENUM_MAX; i++){
-            
-        flag |= (sub_struct.transmit_periods[i] == interval_ms);
-    }
-
-    return flag;
 }
 
-static uint64_t get_interval_ms(void){
+static inline uint64_t get_interval_ms(){
 
-    uint64_t temp_interval_ms;
-
-    if(xSemaphoreTake(sub_struct.interval_mutex, portMAX_DELAY) == pdTRUE){
-
-        temp_interval_ms = sub_struct.interval_ms;
-
-        xSemaphoreGive(sub_struct.interval_mutex);
-    }
-
-    return temp_interval_ms;
+    return sub_struct.transmit_periods[sub_struct.status];
 }
 
-static void run_transmit_timer(void){
-
-    uint64_t temp_interval_ms = get_interval_ms();
-    TickType_t interval_ticks = pdMS_TO_TICKS(temp_interval_ms);
+static inline bool is_sleep_interval(uint64_t interval_ms){
 
-    TickType_t current_timer_period_ticks = xTimerGetPeriod(sub_struct.transmit_timer);
-
-    if(interval_ticks != current_timer_period_ticks && check_transmit_period(temp_interval_ms)){
-
-        xTimerChangePeriod(sub_struct.transmit_timer, pdMS_TO_TICKS(temp_interval_ms), 0);
-    }
-    else {
-
-        xTimerStart(sub_struct.transmit_timer, 0);
-    }   
+    return interval_ms == sub_struct.transmit_periods[SLEEP_MS];
 }
 
-static bool get_transmit_state(void){
+bool get_transmit_state(){
 
-    bool state = false;
-
-    if(xSemaphoreTake(sub_struct.transmit_time_mutex, portMAX_DELAY) == pdTRUE){
-
-        state = sub_struct.transmit_state;
-
-        xSemaphoreGive(sub_struct.interval_mutex);
-    }
-
-    return state;
+    uint64_t current_time_ms = esp_timer_get_time()/1000;
+    
+    return current_time_ms - sub_struct.transmit_timer_ms > get_interval_ms();
 }
 
-static bool is_sleep_interval(uint64_t interval_ms){
-
-    return interval_ms == sub_struct.transmit_periods[SLEEP_MS];
+inline void reset_transmit_timer(){
+    
+    sub_struct.transmit_timer_ms = esp_timer_get_time()/1000;
 }
 
 static void sub_task(void *pvParameters){
 
     while(1){
+
+        check_for_message_ack();
         
         bool state = get_transmit_state();
```

Comment:
Jak dobrze rozumuje to kod nie wyśle paczki jak procek się obudzi ale nadal będzie w stanie do spania. Bo zanim dostanie tutaj true, to już pójdzie spać 😴  Ale mogę się mylić, więc jeszcze sprawdź

Reply:
Tak